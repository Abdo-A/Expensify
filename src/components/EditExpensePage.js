import React from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { editExpense , removeExpense} from '../actions/expenses';

export class EditExpensePage extends React.Component{

    onSubmit=(expense)=>{
        // this.props.dispatch(editExpense(props.match.params.id,expense));
        this.props.makeEditExpense(this.props.expense.id,expense);
        this.props.history.push('/');
    };
    
    onRemove=()=>{
        // this.props.dispatch(removeExpense({id: props.match.params.id}));
        this.props.makeRemoveExpense({id: this.props.expense.id});
        this.props.history.push('/');
    };

    render(){
        return(
            <div>
                <ExpenseForm
                onSubmit={this.onSubmit}
                expense={this.props.expense}
                />
                <button onClick={this.onRemove}>Remove</button>
            </div>
        );
    };
}



const mapStateToProps=(state,props)=>{
    return {
        expense: state.expenses.find((expense)=>{
            return expense.id===props.match.params.id
        })
    };
};

const mapDispatchToProps=(dispatch)=>{
    return{
        makeEditExpense:(id,expense)=>{
            return dispatch(editExpense(id,expense));
        },

        makeRemoveExpense:(id)=>{
            return dispatch(removeExpense(id));
        }
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);