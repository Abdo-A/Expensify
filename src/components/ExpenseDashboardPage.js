import React from 'react';
import numeral from 'numeral';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import getVisibleExpenses from '../selectors/expenses';
import getTotalExpenses from '../selectors/expenses-total';
import { connect } from 'react-redux';


export const ExpenseDashboardPage=(props)=>(
    <div>
        
        <h1>Viewing {props.expenseCount} {props.expenseCount===1?'expense':'expenses'} totalling {props.expensesTotal}</h1>

        <ExpenseListFilters />
        <ExpenseList/>
    </div>
);

const mapStateToProps=(state)=>{
    return{
        expenseCount: getVisibleExpenses(state.expenses,state.filters).length,
        expensesTotal: numeral(getTotalExpenses(getVisibleExpenses(state.expenses,state.filters))/100).format('$0,0.[00]')
    };
};


export default connect(mapStateToProps)(ExpenseDashboardPage);
