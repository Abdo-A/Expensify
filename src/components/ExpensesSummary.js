import React from 'react';
import numeral from 'numeral';
import getVisibleExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import { connect } from 'react-redux';


export const ExpensesSummary=({expenseCount, expensesTotal})=>(
    <div>
        <h1>Viewing {expenseCount} {expenseCount===1?'expense':'expenses'} totalling {expensesTotal}</h1>
    </div>
);


const mapStateToProps=(state)=>{
    const visibleExpenses=getVisibleExpenses(state.expenses,state.filters);
    return{
        expenseCount: visibleExpenses.length,
        expensesTotal: numeral(selectExpensesTotal(visibleExpenses)/100).format('$0,0.[00]')
    };
};


export default connect(mapStateToProps)(ExpensesSummary);