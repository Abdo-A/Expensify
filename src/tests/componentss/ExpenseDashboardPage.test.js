import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseDashboardPage } from '../../components/ExpenseDashboardPage';
import testExpenses from '../fixtures/expenses';

test('should render ExpenseDashboardPage correctly',()=>{
    const wrapper = shallow(<ExpenseDashboardPage/>);
    expect(wrapper).toMatchSnapshot();
});

test('should view a single expense correctly',()=>{
    const wrapper = shallow(<ExpenseDashboardPage expenseCount={1} expensesTotal={'$94.34'}/>);
    expect(wrapper).toMatchSnapshot();
});


test('should view a more than one expense correctly',()=>{
    const wrapper = shallow(<ExpenseDashboardPage expenseCount={2} expensesTotal={'$94.34'}/>);
    expect(wrapper).toMatchSnapshot();
});