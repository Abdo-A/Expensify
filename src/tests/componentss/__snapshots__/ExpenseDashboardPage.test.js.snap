// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseDashboardPage correctly 1`] = `
<div>
  <h1>
    Viewing 
     
    expenses
     totalling 
  </h1>
  <Connect(ExpenseListFilters) />
  <Connect(ExpenseList) />
</div>
`;

exports[`should view a more than one expense correctly 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totalling 
    $94.34
  </h1>
  <Connect(ExpenseListFilters) />
  <Connect(ExpenseList) />
</div>
`;

exports[`should view a single expense correctly 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $94.34
  </h1>
  <Connect(ExpenseListFilters) />
  <Connect(ExpenseList) />
</div>
`;
